src/components/mortgage-investment-compare/index.js

```diff
@@ -0,0 +1,220 @@
+  calculateAll (state) {
+    option1 = state.options.option1
+    option2 = state.options.option2
+
+    option1.pmt = this.calculatePMT(option1, state)
+    option2.pmt = this.calculatePMT(option2, state)
+
+    option1.interestAmt = this.calculateInterestAmt(option1, state)
+    option2.interestAmt = this.calculateInterestAmt(option2, state)
+
+    option1.fv = this.calculateFV(option1, option2, state)
+    option2.fv = this.calculateFV(option2, option1, state)
+
+    state.optCost = this.calculateOpportunityCost(option1, option2)
+
+    const [shorter, longer] = _.sortBy([option1, option2], 'term')
+    state.shorterOption = shorter.term === option1.term ? 'option1' : 'option2'
+    state.longerOption = longer.term === option1.term ? 'option1' : 'option2'
+
+    state.yearlyResultsByOption = getYearly({
+      loanAmt: state.loanAmt,
+      investmentRate: state.investmentRate / 100,
+      inflationRate: state.inflation / 100,
+      shorterOption: {
+        mortgageRate: shorter.interestRate / 100,
+        mortgageTerm: shorter.term
+      },
+      longerOption: {
+        mortgageRate: longer.interestRate / 100,
+        mortgageTerm: longer.term
+      }
+    })
+
+    return state
+  }
+
+  render () {
+    const shorterOption = this.state.options[this.state.shorterOption]
+    const longerOption = this.state.options[this.state.longerOption]
+    const bestOption = _.maxBy([shorterOption, longerOption], 'fv')
+
+    return (
+      <Row>
+        <Col xs='4'>
+          <InputCard
+            loanAmt={this.state.loanAmt}
+            inflation={this.state.inflation}
+            investmentRate={this.state.investmentRate}
+            updateInput={this.updateInput}
+            option1={option1}
+            option2={option2}
+            shorterOption={shorterOption}
+            longerOption={longerOption}
+          />
+        </Col>
+        <Col xs='8'>
+          <Summary
+            shorterOption={shorterOption}
+            longerOption={longerOption}
+            bestOption={bestOption}
+            optCost={this.state.optCost}
+            inflation={this.state.inflation}
+            investmentRate={this.state.investmentRate}
+            yearlyResultsByOption={this.state.yearlyResultsByOption}
+          />
+          <Button
+            className='d-block mx-auto mt-4'
+            outline
+            color='success'
+            onClick={this.toggleShowTable}
+          >
+            {this.state.showTable ? 'Hide' : 'See'} Yearly Breakdown
+          </Button>
+          <AmortizationTable
+            option1={option1}
+            option2={option2}
+            shorterOption={shorterOption}
+            longerOption={longerOption}
+            yearlyResultsByOption={this.state.yearlyResultsByOption}
+            showTable={this.state.showTable}
+          />
+          {
+            <pre>
+              {JSON.stringify(this.state, null, 4).replace(/[{}]/g, '')}
+            </pre>
+          }
```

Comment:
q: Are these outer `{...}` needed?

Reply:
I believe so, they close the react class don't they?

Reply:
they are not, I can delete them